### PR TITLE
PHOENIX-6929 Explicitly depend on org.glassfish:javax.el

### PR DESCRIPTION
**Pull request by shahrs87**

### CODE REVIEW

**Comment by shahrs87** [comment]
@virajjasani  Please review.

**Review by virajjasani — APPROVED** [review]

**Comment by gjacoby126** [comment]
@shahrs87 - is this the correct version to use? Latest version of Glassfish's javax.el looks like 3.0-b12, and there's a note in mvnrepository that it's been deprecated in favor of jakarta.el. 

In addition the glassfish el jars seem to have CVEs attached. 
https://mvnrepository.com/artifact/org.glassfish/javax.el/3.0.1-b12

I'm not familiar enough with the Java EE expression language framework here to know the correct answer.

**Review by gjacoby126 — CHANGES_REQUESTED** [review]
Questions about what the correct version is

**Comment by virajjasani** [comment]
@gjacoby126 this was mostly to stay in sync with hbase https://github.com/apache/hbase/blob/master/pom.xml#L826

**Comment by shahrs87** [comment]
> is this the correct version to use? Latest version of Glassfish's javax.el looks like 3.0-b12

I can update it to 3.0-b12.

> and there's a note in mvnrepository that it's been deprecated in favor of jakarta.el.

I am not inclined to change to jakarta.el because hbase-server will try to download `org.glassfish:javax.el:jar:3.0.1-b06-SNAPSHOT`. To avoid that, I would have to exclude `org.glassfish:javax.el:jar` from multiple module's pom.

> In addition the glassfish el jars seem to have CVEs attached.

The [linked CVE](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-15250) affects junit version from version 4.7 and before 4.13.1. Looking at the mvn dependency tree for junit:junit, we are including junit version 4.13.1 which is free from this vulnerability.
```
mvn dependency:tree -Dincludes=junit:junit

[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ phoenix-hbase-compat-2.5.0 ---
[INFO] org.apache.phoenix:phoenix-hbase-compat-2.5.0:jar:5.2.0-SNAPSHOT
[INFO] \- junit:junit:jar:4.13.1:test
[INFO] 
[INFO] -----------< org.apache.phoenix:phoenix-hbase-compat-2.4.1 >------------
[INFO] Building Phoenix Hbase 2.4.1 compatibility 5.2.0-SNAPSHOT         [3/11]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ phoenix-hbase-compat-2.4.1 ---
[INFO] org.apache.phoenix:phoenix-hbase-compat-2.4.1:jar:5.2.0-SNAPSHOT
[INFO] \- junit:junit:jar:4.13.1:test
[INFO] 
[INFO] -----------< org.apache.phoenix:phoenix-hbase-compat-2.4.0 >------------
[INFO] Building Phoenix Hbase 2.4.0 compatibility 5.2.0-SNAPSHOT         [4/11]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ phoenix-hbase-compat-2.4.0 ---
[INFO] org.apache.phoenix:phoenix-hbase-compat-2.4.0:jar:5.2.0-SNAPSHOT
[INFO] \- junit:junit:jar:4.13.1:test
[INFO] 
[INFO] ------------------< org.apache.phoenix:phoenix-core >-------------------
[INFO] Building Phoenix Core 5.2.0-SNAPSHOT                              [5/11]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ phoenix-core ---
[INFO] org.apache.phoenix:phoenix-core:jar:5.2.0-SNAPSHOT
[INFO] \- junit:junit:jar:4.13.1:test
[INFO] 
[INFO] ------------------< org.apache.phoenix:phoenix-pherf >------------------
[INFO] Building Phoenix - Pherf 5.2.0-SNAPSHOT                           [6/11]
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for junit:junit-dep:jar:4.9.1-SNAPSHOT is missing, no dependency information available
[INFO] 
[INFO] --- maven-dependency-plugin:3.1.1:tree (default-cli) @ phoenix-pherf ---
[INFO] org.apache.phoenix:phoenix-pherf:jar:5.2.0-SNAPSHOT
[INFO] \- junit:junit:jar:4.13.1:test
```

@gjacoby126  Let me know if you still have concerns. Thank you.

**Comment by virajjasani** [comment]
I am not sure if explicitly using different version than the one used by hbase might have any implications with any of our server side workflows

**Comment by gjacoby126** [comment]
@virajjasani @shahrs87 - good point about syncing up with HBase. We should use what they use for now. (But I wonder if HBase also needs to upgrade...when I was trying to get Phoenix and Dropwizard to play nicely a year or so back, these obsolete javax / Jakarta EE jars made things painful.)

**Comment by virajjasani** [comment]
> But I wonder if HBase also needs to upgrade

+1

**Comment by shahrs87** [comment]
There is 1 test failure: `org.apache.phoenix.jdbc.ParallelPhoenixUtilTest#getAnyOfNonExceptionallyTimeoutTest`. It is a known flaky test. It runs fine locally.
Build link: https://ci-hadoop.apache.org/blue/organizations/jenkins/Phoenix%2FPhoenix-PreCommit-GitHub-PR/detail/PR-1591/4/tests/
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.apache.phoenix.jdbc.ParallelPhoenixUtilTest
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.095 s - in org.apache.phoenix.jdbc.ParallelPhoenixUtilTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

**Comment by shahrs87** [comment]
@virajjasani  @gjacoby126  Can you please review and merge the change? Thank you.

File: pom.xml [hunk 1]
```diff
@@ -169,6 +169,7 @@
     <!-- PHOENIX-6806 - Default protoc architecture to the detected one,
       but allow for profile overrides -->
     <protoc.arch>${os.detected.classifier}</protoc.arch>
+    <glassfish.el.version>3.0.1-b08</glassfish.el.version>
```
Comment (virajjasani) [inline pom.xml:172]: Maybe good to add a comment here "keep the version in sync with hbase pom's javax.el version"?

**Comment by virajjasani** [comment]
@gjacoby126 you are good with the changes? or shall we create Jira in hbase first?

**Review by gjacoby126 — APPROVED** [review]
+1

**Pull request merged**